skops/card/_model_card.py

```diff
@@ -288,64 +438,373 @@ class Card:
+            dict).
+
+        Raises
+        ------
+        KeyError
+            If the given section name was not found, a ``KeyError`` is raised.
+
+        """
+        if not key:
+            msg = f"Section name cannot be empty but got '{key}'"
+            raise KeyError(msg)
+
+        *subsection_names, leaf_node_name = split_subsection_names(key)
+
+        if not leaf_node_name:
+            msg = f"Section name cannot be empty but got '{key}'"
+            raise KeyError(msg)
+
+        parent_section = self._select(subsection_names, create=False)
+        return parent_section[leaf_node_name]
+
+    def delete(self, key: str | Sequence[str]) -> None:
+        """Delete a section from the model card.
+
+        To delete a subsection of an existing section, use a ``"/"`` in the
+        section name, e.g.:
+
+        ``card.delete("Existing section/New section")``.
+
+        Alternatively, a list of strings can be passed:
+
+        ``card.delete(["Existing section", "New section"])``.
+
+        Parameters
+        ----------
+        key : str or list of str
+            The name of the (sub)section to select. When selecting a subsection,
+            either use a ``"/"`` in the name to separate the parent and child
+            sections, or pass a list of strings.
+
+        Raises
+        ------
+        KeyError
+            If the given section name was not found, a ``KeyError`` is raised.
+
+        """
+        if not key:
+            msg = f"Section name cannot be empty but got '{key}'"
+            raise KeyError(msg)
+
+        if isinstance(key, str):
+            *subsection_names, leaf_node_name = split_subsection_names(key)
+        else:
+            *subsection_names, leaf_node_name = key
+
+        if not leaf_node_name:
+            msg = f"Section name cannot be empty but got '{key}'"
+            raise KeyError(msg)
+
+        parent_section = self._select(subsection_names, create=False)
+        del parent_section[leaf_node_name]
+
+    def _add_single(self, key: str, val: Formattable | str) -> Section:
+        """Add a single section.
+
+        If the (sub)section does not exist, it is created. Otherwise, the
+        existing (sub)section is modified.
+
+        Parameters
+        ----------
+        key: str
+            The name of the (sub)section.
+
+        val: str or Formattable
+            The value to assign to the (sub)section.
+
+        """
+        *subsection_names, leaf_node_name = split_subsection_names(key)
+        section = self._select(subsection_names)
+
+        if leaf_node_name in section:
+            # entry exists, only overwrite content
+            section[leaf_node_name].content = val
+        else:
+            # entry does not exist, create a new one
+            section[leaf_node_name] = Section(title=leaf_node_name, content=val)
+
+        return section[leaf_node_name]
+
+    def _add_model_section(self, model) -> None:
```

Comment:
we talked about exposing these somehow to users with custom templates, how is that done now?

Reply:
The thing with these methods is tat they do more than a simple `add`. So what we could do in this PR, is to make them public and change the signature to:

```py
    def add_model_section(self, model, section=None) -> None:
```

Then we'd only accept `section=None` if template is the default template.

Reply:
I don't remember that we discussed it like that, but I can take a look :)

Presumably, we don't need the `model` argument and use `self.model` instead? Should users be allowed to add the model reprs multiple times into different sections?

Reply:
if we can use `self.model`, then this argument shouldn't be here in the first place.

I don't care for now if users end up adding it multiple times. If they do, their model card looks silly.